src/Avalonia.Base/ValueStore.cs

```diff
@@ -103,7 +103,7 @@ public bool IsSet(AvaloniaProperty property)
 
             IDisposable? result = null;
 
-            if (_values.TryGetValue(property, out var slot))
+            if (_values.TryGetValue(property, out var slot) && !IsRemoveSentinel<T>(slot))
```

Comment:
I'm still not 100% sure why only certain places respect sentinel values, perhaps we should introduce a small helper to get values (with or without sentinel value handling). And perhaps a comment explaining why we need to handle sentinel values in certain cases would help with maintenance.

Reply:
Good point, I've made handling sentinel values more consistent.